pkg/manager/manager.go

```diff
@@ -475,7 +458,8 @@ func (m *manager) doCommitAll(config globalSpec) error {
 				log.Error("Cannot commit group", "spec", spec, "err", err)
 			}
 			return true, err
-		})
+		},
+		true)
```

Comment:
Can you make this `true` a parameter of this `doCommitAll()` function?  Then in the code near `onAssumeLeadership` we pass this `true` explicitly and then in a regular `Commit` we pass false?  Then we can add a comment on the meaning of issuing two commits on leadership failover.  This should make the logic easier to follow.  Thoughts?

Reply:
Well, `doCommitAll` is only invoked from  `loadAndCommitSpecs` which is only invoked from `onAssumeLeadership`. So, if we want to make it more _explicit_ on `onAssumeLeadership` then we need to pass this new `bool` through. But, since a "commit" only happens when leadership is assumed then it's this already intuitive?

I'm all for changing the flow (since it took me a long time to understand this code) but I don't know if this proposed change will do it.

The motivation behind this additional param was to do the 2nd group commit after all of the groups were originally committed. In theory, we could instead just call commit 2 times in a row per group. Thoughts?

Reply:
Yup I agree with you.  Can you add a comment at the `true` to mention it's needed to continue updating in the event of failover?

Reply:
Yep, see new commit.